File: chess_java/src/chess/piecies/Rook.java
package chess.piecies;

import chess.Board;
import chess.Spot;
import chess.utils.PathCalculator;

import java.util.List;

public class Rook extends Piece{

    private boolean isMoved;

    public Rook(boolean white) {
        super(white);
    }


    @Override
    public List<Spot> getPathSpots(Board board, Spot start, Spot end) {
       return PathCalculator.getLinearPathSpots(board, start, end);
    }


    public boolean isMoved() {
        return isMoved;
    }

    public void setMoved(boolean moved) {
        isMoved = moved;
    }
}

File: chess_cpp/piecies/Rook.cpp
//
// Created by Hrach Khachatryan on 12/15/19.
//

#include "Rook.h"

Rook::Rook(bool white) : Piece(white) {
}

bool Rook::isMoved() {
    return moved;
}

void Rook::setMoved(bool moved) {
    this->moved = moved;
}

std::vector<Spot> Rook::getPathSpots(Board *board, Spot *start, Spot *end) {
    return PathCalculator::getLinearPathSpots(board, start, end);
}
